portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,27 +14,66 @@
 
 package com.google.sps.servlets;
 
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.gson.Gson;
+import com.google.sps.data.Comment;
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
-import java.util.ArrayList;
-import com.google.gson.Gson;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-  private final ArrayList<String> msgs = new ArrayList<String>();
 
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    msgs.add("hello");
-    msgs.add("there");
-    msgs.add("world");
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {    
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    int count = 1;
+    List<Comment> entries = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+      if (count <= 5){
```

Comment:
use an early break instead of iterating through every value:
```
for(entity : results) {
  if (count >= 5) {
    break;
  }
  ...
}
```

Reply:
If I do that, then no comments get displayed if I have say 6 or more comments (because none of them get added to array list!) So, I am not implementing this change.

Reply:
You should be able to just move everything currently inside your if statement below the if statement Nick is proposing, so the first 5 comments will still go through that segment of code, and you'll keep incrementing 'count', but once 'count' reaches 5 you'll break out of the for loop before adding a 6th comment to the array. So like: 

```
    int count = 0;
    List<Comment> entries = new ArrayList<>();
    for (Entity entity : results.asIterable()) {
        if (count >= 5) {
            break;
        }
        long id = entity.getKey().getId();
        .
        .
        .
        entries.add(entry);
        count++;
    }
```

Reply:
Ok, it works now, not sure why it didn't work before!